### PR TITLE
Table cell bool rendering

### LINKED ISSUE #662: BUG-The display of multiple Bools in tables slows down the application
A table visual element will display boolean values like this:
image

This looks great but unfortunately, when you have 100 rows and 10 columns of booleans, 1000 of these toggles are rendered on the page which slows down the application to the point where every interaction takes some time to be processed.

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
commanded by light_bool_render
resolves #662

```
import pandas as pd

from taipy.gui import Gui

value = 10
data = pd.DataFrame()

# Create 1O columns with 100 rows of bools
for i in range(10):
    data[i] = [True if j % value == 0 else False for j in range(100)]

page = """
<|{value}|slider|>
<|{data}|table|use_checkbox|lov[0]=Male;Female|lov[1]=0;1|lov[2]=Bad;Good|lov[3]=Disagree;Agree|>
"""

Gui(page).run(title="662 table cell boolean render")

```

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx [hunk 1]
```diff
@@ -94,6 +94,7 @@ interface RowData {
     lineStyle?: string;
     nanValue?: string;
     compRows?: RowType[];
+    renderLightBool?: boolean;
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx]: `useCheckbox`? `checkbox` (😒)?

File: frontend/taipy-gui/src/components/Taipy/tableUtils.tsx [hunk 2]
```diff
@@ -472,14 +475,27 @@ export const EditableCell = (props: EditableCellProps) => {
                 {edit ? (
                     colDesc.type?.startsWith("bool") ? (
                         <Box sx={cellBoxSx}>
+                            lightBool ? (
+                            <input
+                                type="checkbox"
+                                checked={val as boolean}
+                                title={val ? "True" : "False"}
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/tableUtils.tsx]: What about using the property as the place to define those texts?
Like... `|use_checkbox=False;True|` or
```
check_values = ("Off", "On")
|use_checkbox={check_values}|
```
If `|use_checkbox|` (`tgb.table(..., use_checkbox=True)`) then False/True would be the 'default`...

File: taipy/gui/viselements.json [hunk 3]
```diff
@@ -965,6 +966,12 @@
                                 "list[str]"
                             ]
                         ]
+                    },
+                    {
+                        "name": "render_light_bool",
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]: to be continued...

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
What do you think?

File: frontend/taipy-gui/src/components/Taipy/tableUtils.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/tableUtils.tsx]: those 2 are not related
the title is defined on the switch control too

File: frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx]: use_checkbox then

File: frontend/taipy-gui/src/components/Taipy/tableUtils.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/tableUtils.tsx]: [#1826](https://github.com/Avaiga/taipy/issues/1826)

File: frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx [hunk 4]
```diff
@@ -555,10 +559,12 @@ const AutoLoadingTable = (props: TaipyTableProps) => {
             lineStyle: props.lineStyle,
             nanValue: props.nanValue,
             compRows: compRows,
+            renderLightBool: useCheckbox,
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx]: ?

**Review by FabienLelaquais — DISMISSED** [review]
I LoV it!

File: frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx (hunk 4 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx]: good catch

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx]: and the checker does not rale !

**Review by FabienLelaquais — DISMISSED** [review]
👍

**Review by FabienLelaquais — APPROVED** [review]
I like those changes!

**Pull request merged**